_search-plugins/sql/limitation.md

```diff
@@ -29,10 +29,24 @@ FROM (
 
 But, if the outer query has `GROUP BY` or `ORDER BY`, then it's not supported.
 
-## JOIN does not support aggregations on the joined result
+## JOIN queries
 
-The `join` query does not support aggregations on the joined result.
-For example, e.g. `SELECT depo.name, avg(empo.age) FROM empo JOIN depo WHERE empo.id == depo.id GROUP BY depo.name` is not supported.
+Because OpenSearch doesn't natively support relational operations, `JOIN` queries are supported on a best-effort basis.
+
+### JOIN does not support aggregations on the joined result
+
+The `JOIN` query does not support aggregations on the joined result.
+For example, `SELECT depo.name, avg(empo.age) FROM empo JOIN depo WHERE empo.id = depo.id GROUP BY depo.name` is not supported.
+
+### Performance
+
+`JOIN` queries are prone to expensive index scanning operations.
+Depending on the dataset, there may be scalability issues when running `JOIN` queries between data sets with more than around 5 million records.
+
+To avoid runaway resource usage, `JOIN` queries will time out after 60 seconds.
+This limit is currently not configurable.
```

Comment:
This is not true? I can see a SQL hint for this: https://github.com/opensearch-project/sql/blob/d9b5f2d9ca5f97273f043100f2dbbf4c5abd44a7/legacy/src/test/java/org/opensearch/sql/legacy/unittest/planner/QueryPlannerMonitorTest.java#L73. Probably it's just not documented.

Reply:
Huh, TIL. I was looking at the [hash join executor impl](https://github.com/opensearch-project/sql/blob/d9b5f2d9ca5f97273f043100f2dbbf4c5abd44a7/legacy/src/main/java/org/opensearch/sql/legacy/query/planner/HashJoinQueryPlanRequestBuilder.java#L34), but I guess the hint operates at a different abstraction level.